backend/src/db/providers/clientCourseProvider.ts

```diff
@@ -67,7 +67,11 @@ const getClientCourseProvider = async (clientCourseId: string): Promise<IClientC
   return clientCourse;
 };
 
-const applyCourseProvider = async (courseId: string, userId: string, progressDto: IProgress[]) => {
+const applyCourseProvider = async (
+  courseId: string,
+  userId: string,
+  progressDto: IProgress[],
+): Promise<mongoose.Document<Types.ObjectId, IClientCourse, IClientCourse> & IClientCourse> => {
```

Comment:
No `mongoose.Document` pls. As far as I understand this provider returns common ClientCourse Object